src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECParameters.cs

```diff
@@ -62,5 +66,374 @@ public void Validate()
+
+            writer.PopSequence();
+            return writer;
+        }
+
+        private AsnWriter WriteAlgorithmIdentifier()
+        {
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+            WriteAlgorithmIdentifier(writer);
+            return writer;
+        }
+
+        private void WriteAlgorithmIdentifier(AsnWriter writer)
+        {
+            writer.PushSequence();
+
+            writer.WriteObjectIdentifier(Oids.EcPublicKey);
+            WriteEcParameters(writer);
+
+            writer.PopSequence();
+        }
+
+        internal AsnWriter WritePkcs8PrivateKey()
+        {
+            Validate();
+
+            if (D == null)
+            {
+                throw new CryptographicException(SR.Cryptography_CSP_NoPrivateKey);
+            }
+
+            if (!Curve.IsNamed)
+            {
+                throw new CryptographicException(SR.Cryptography_ECC_NamedCurvesOnly);
+            }
+
+            // Don't need the domain parameters because they're contained in the algId.
+            using (AsnWriter ecPrivateKey = WriteEcPrivateKey(includeDomainParameters: false))
+            using (AsnWriter algorithmIdentifier = WriteAlgorithmIdentifier())
+            {
+                return KeyFormatHelper.WritePkcs8(algorithmIdentifier, ecPrivateKey);
+            }
+        }
+
+        private void WriteEcParameters(AsnWriter writer)
+        {
+            if (Curve.IsNamed)
+            {
+                Oid oid = Curve.Oid;
+
+                // On Windows the FriendlyName is populated in places where the Value mightn't be.
+                if (string.IsNullOrEmpty(oid.Value))
+                {
+                    oid = Oid.FromFriendlyName(oid.FriendlyName, OidGroup.All);
+                }
+
+                writer.WriteObjectIdentifier(oid.Value);
+            }
+            else
+            {
+                throw new CryptographicException(SR.Cryptography_ECC_NamedCurvesOnly);
+            }
+        }
+
+        private void WriteUncompressedPublicKey(AsnWriter writer)
+        {
+            int publicKeyLength = Q.X.Length * 2 + 1;
+            Span<byte> publicKeyBytes = stackalloc byte[0];
+            byte[] publicKeyRented = null;
+
+            if (publicKeyLength < 256)
+            {
+                publicKeyBytes = stackalloc byte[publicKeyLength];
+            }
+            else
+            {
+                publicKeyRented = ArrayPool<byte>.Shared.Rent(publicKeyLength);
+                publicKeyBytes = publicKeyRented.AsSpan(0, publicKeyLength);
+            }
+
+            try
+            {
+                publicKeyBytes[0] = 0x04;
+                Q.X.AsSpan().CopyTo(publicKeyBytes.Slice(1));
+                Q.Y.AsSpan().CopyTo(publicKeyBytes.Slice(1 + Q.X.Length));
+
+                writer.WriteBitString(publicKeyBytes);
+            }
+            finally
+            {
+                CryptographicOperations.ZeroMemory(publicKeyBytes);
```

Comment:
why do we care about clearing public key?

Reply:
Just tidiness.  No reason to let it get fished back out of the array pool.